packages/react/src/components/Toolbar/ToolbarMenuItem.tsx

```diff
@@ -176,7 +338,37 @@ class ToolbarMenuItem extends UIComponent<WithAsProp<ToolbarMenuItemProps>> {
       })
     }
 
-    const menuItemInner = hasChildren ? children : elementType
+    const menuItemInner = hasChildren ? children : <Ref innerRef={this.itemRef}>{elementType}</Ref>
+
+    const maybeSubmenu =
+      menu && menuOpen ? (
+        <Unstable_NestingAuto>
+          {(getRefs, nestingRef) => (
+            <>
+              <Ref innerRef={nestingRef}>
+                <Ref innerRef={this.menuRef}>
```

Comment:
Oops, you don't need two `Ref` components:

```tsx
<Ref innerRef={(node: HTMLElement) => {
  nestingRef.current = node
  this.menuRef.current = node
}} />
```

To make it assignable on `this.menuRef`:

```tsx
menuRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>
```